source/includes/write/bulk-write.java

```diff
@@ -0,0 +1,128 @@
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.ServerApi;
+import com.mongodb.ServerApiVersion;
+
+import com.mongodb.bulk.BulkWriteResult;
+import com.mongodb.client.model.InsertOneModel;
+import com.mongodb.client.model.UpdateOneModel;
+import com.mongodb.client.model.UpdateManyModel;
+import com.mongodb.client.model.ReplaceOneModel;
+import com.mongodb.client.model.DeleteOneModel;
+import com.mongodb.client.model.DeleteManyModel;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.reactivestreams.client.MongoCollection;
+
+import org.bson.Document;
+
+import com.mongodb.reactivestreams.client.MongoClient;
+import com.mongodb.reactivestreams.client.MongoClients;
+import com.mongodb.reactivestreams.client.MongoDatabase;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+import java.util.Arrays;
+
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Updates.set;
+
+class BulkWrite {
+    public static void main(String[] args) throws InterruptedException {
+        // Replace the placeholder with your Atlas connection string
+        String uri = "<connection string>";
+
+        // Construct a ServerApi instance using the ServerApi.builder() method
+        ServerApi serverApi = ServerApi.builder()
+                .version(ServerApiVersion.V1)
+                .build();
+
+        MongoClientSettings settings = MongoClientSettings.builder()
+                .applyConnectionString(new ConnectionString(uri))
+                .serverApi(serverApi)
+                .build();
+
+        // Create a new client and connect to the server
+        try (MongoClient mongoClient = MongoClients.create(settings)) {
+            MongoDatabase sample_restaurants = mongoClient.getDatabase("sample_restaurants");
+            MongoCollection<Document> restaurants = sample_restaurants.getCollection("restaurants");
+
+
+            // start-bulk-insert-one
+            InsertOneModel<Document> operation = new InsertOneModel<>(
+                    new Document("name", "Mongo's Deli")
+                            .append("cuisine", "Sandwiches"));
+            // end-bulk-insert-one
+
+            // start-bulk-update-one
+            UpdateOneModel<Document> operation = new UpdateOneModel<>(
+                    eq("name", "Mongo's Deli"),
+                    set("cuisine", "Sandwiches and Salads"));
+            // end-bulk-update-one
+
+            // start-bulk-update-many
+            UpdateManyModel<Document> operation = new UpdateManyModel<>(
+                    eq("name", "Mongo's Deli"),
+                    set("cuisine", "Sandwiches and Salads"));
+            // end-bulk-update-many
+
+            // start-bulk-replace-one
+            ReplaceOneModel<Document> operation = new ReplaceOneModel<>(
+                    eq("restaurant_id", "1234"),
+                    new Document("name", "Mongo's Pizza"));
```

Comment:
This example seems a little artificial, as it replaces the whole document you'd lose the `restaurant_id` is that desired?

Reply:
Will tweak the example to something in line with the example in [the server manual](https://www.mongodb.com/docs/manual/reference/method/db.collection.replaceOne/).